src/main/java/dev/dfonline/codeclient/command/CommandSender.java

```diff
@@ -0,0 +1,61 @@
+package dev.dfonline.codeclient.command;
+
+import dev.dfonline.codeclient.CodeClient;
+import dev.dfonline.codeclient.RateLimiter;
+
+import java.util.ArrayDeque;
+
+/**
+ * Queues up commands and sends them to avoid getting kicked for spam.
+ */
+public class CommandSender {
+    // Vanilla Minecraft uses 20 increment 200 threshold.
+    // We have a lower threshold for extra safety and to account for lag.
+    private static final RateLimiter rateLimiter = new RateLimiter(20, 140);
+    private static final ArrayDeque<String> commandQueue = new ArrayDeque<>();
+
+    /*
+    public static void queue(String command, int delay, Runnable callback) {
+        rateLimiter.increment();
+        for (int i = 0; i < delay; i++) {
+        }
+        commandQueue.add(command);
+        if (callback != null) {
+            callback.run();
+        }
+    }
+
+    public static void queue(String command, int delay) {
+        queue(command, delay, null);
+    }
+
+    public static void queue(String command) {
+        queue(command, 0, null);
+    }
+    */
+
+    public static void queue(String command) {
+        rateLimiter.increment();
```

Comment:
You don't need to rate limit adding to the queue, it should only register an operation on the tick method where the commands are actually sent, if you'd like you could make an internal rate limiter but I don't think that's necessary

Reply:
ah gotcha, i was just copying what you sent me :P